pikachewie/publisher.py

```diff
@@ -51,6 +51,20 @@ def publish(self, exchange, routing_key, body, properties=None):
                 body=body,
             )
 
+    def process_data_events(self, time_limit=0):
+        """Calls process_data_events on the current connection (if there is
+        currently one open).
+        """
+        if self._channel:
```

Comment:
I can't see a way of referencing the connection if there's no channel available - if there was, then I would have accessed it directly. If there's no `_channel` attribute, doesn't that mean there's no connection object that can be referenced?